tests/aws/services/events/test_events_targets.py

```diff
@@ -34,6 +36,273 @@
+                    "Action": "execute-api:Invoke",
+                    "Resource": f'arn:aws:execute-api:{apigateway_client.meta.region_name}:{sts_client.get_caller_identity()["Account"]}:{api_id}/*/*/*'
+                }
+            ]
+        })
+
+        iam_role_name = f"EventBridge_Invoke_API_{short_uid()}"
+
+        # Use the existing 'create_role' fixture
+        create_role_response = create_role(
+            RoleName=iam_role_name,
+            AssumeRolePolicyDocument=assume_role_policy_document
+        )
+        role_arn = create_role_response['Role']['Arn']
+
+        iam_client.put_role_policy(
+            RoleName=iam_role_name,
+            PolicyName='InvokeApiPolicy',
+            PolicyDocument=policy_document
+        )
+
+        # Allow some time for IAM role propagation (only needed in AWS)
+        if is_aws_cloud():
+            time.sleep(10)
+
+        # Step e: Add the API Gateway as a target with the RoleArn
+        target_id = f"target-{short_uid()}"
+
+        # Construct the API target ARN
+        api_target_arn = f'arn:aws:execute-api:{apigateway_client.meta.region_name}:{sts_client.get_caller_identity()["Account"]}:{api_id}/{stage_name}/POST/test'
+
+        put_targets_response = events_client.put_targets(
+            Rule=rule_name,
+            EventBusName=event_bus_name,
+            Targets=[
+                {
+                    "Id": target_id,
+                    "Arn": api_target_arn,
+                    "RoleArn": role_arn,
+                    "Input": json.dumps({"message": "Hello from EventBridge"}),
+                    "RetryPolicy": {
+                        'MaximumRetryAttempts': 0 
+                    }
+                }
+            ]
+        )
+        snapshot.match("put_targets_response", put_targets_response)
+        assert put_targets_response['FailedEntryCount'] == 0
+
+        # Step f: Send an event to EventBridge
+        import uuid
+
+        event_entry = {
+            "EventBusName": event_bus_name,
+            "Source": "test.source",
+            "DetailType": "test.detail.type",
+            "Detail": json.dumps({"message": "Hello from EventBridge"})
+        }
+        put_events_response = events_client.put_events(
+            Entries=[event_entry]
+        )
+        snapshot.match("put_events_response", put_events_response)
+        assert put_events_response['FailedEntryCount'] == 0
+
+        # Step g: Verify the Lambda invocation
+        time.sleep(30) 
+        try:
+            events = retry(
+                check_expected_lambda_log_events_length,
+                retries=10,
+                sleep=10,
+                function_name=function_name,
+                expected_length=1,
+                logs_client=logs_client,
+            )
+            snapshot.match("lambda_logs", events)
+        except Exception as e:
+            pytest.fail(f"Lambda invocation verification failed: {e}")
+
+        # Step h: Clean up resources
```

Comment:
Most of the fixtures I suggested should do this resource cleaning up for you which should clear up created resources on teardown of a test. So in the event that your test fails prior to reaching the below statements, your resources will not be cleared up.

For those resources not cleaned up via fixtures (pretty safe bet that most fixtures do cleanups), we have another `cleanups` fixture you can use to enact all these deletions on teardown -- meaning an early exit will still clear things up on AWS.

See fixture:
https://github.com/localstack/localstack/blob/b770a274ea3b9bde170b802686da23d2a644dc3f/localstack-core/localstack/testing/pytest/fixtures.py#L1891

See usage (note that the delete operation takes place in a `lambda`, allowing for lazy execution):
https://github.com/localstack/localstack/blob/b770a274ea3b9bde170b802686da23d2a644dc3f/tests/aws/test_integration.py#L272